modules/kinesis/src/it/scala/com/snowplowanalytics/snowplow/enrich/kinesis/EnrichKinesisSpec.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2022-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.enrich.kinesis
+
+import scala.concurrent.duration._
+
+import cats.effect.{IO, Resource}
+
+import retry.syntax.all._
+import retry.RetryPolicies
+
+import cats.implicits._
+
+import cats.effect.testing.specs2.CatsIO
+
+import org.testcontainers.containers.{GenericContainer => JGenericContainer}
+
+import org.specs2.mutable.Specification
+import org.specs2.specification.AfterAll
+
+class EnrichKinesisSpec extends Specification with AfterAll with CatsIO {
+
+  override protected val Timeout = 10.minutes
+
+  val retryPolicy =
+    RetryPolicies.limitRetriesByCumulativeDelay(5.minutes, RetryPolicies.capDelay[IO](2.second, RetryPolicies.fullJitter[IO](1.second)))
+
+  def afterAll: Unit = Containers.localstack.stop()
+
+  "enrich-kinesis" should {
+    "initialize with the minimal config" in {
+      Resource.make[IO, JGenericContainer[_]](
```

Comment:
Can we hide `Resource` stuff in the `enrich` method? So `enrich` could return `Resource[IO, Container]`? 

Then in a test: 

```scala

"TEST" in {

gimmeMeEnrich(
     config = "This config",
     app/loggerName = "Enrich is named like that in logs"
).use { instance => 
    // do something
}